tests/Domain/Insights/ForbiddenFinalClassTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Domain\Insights;
+
+use NunoMaduro\PhpInsights\Domain\Analyser;
+use NunoMaduro\PhpInsights\Domain\Insights\ForbiddenFinalClasses;
+use NunoMaduro\PhpInsights\Domain\Metrics\Architecture\Classes;
+use Tests\TestCase;
+
+final class ForbiddenFinalClassTest extends TestCase
+{
+    public function testHasIssue(): void
+    {
+        $files = [
+            __DIR__ . '/Fixtures/ForbiddenFinalClass.php',
+        ];
+
+        $analyzer = new Analyser();
+        $collector = $analyzer->analyse(__DIR__ . '/Fixtures/', $files);
+        $insight = new ForbiddenFinalClasses($collector, []);
+
+        self::assertTrue($insight->hasIssue());
+        self::assertIsArray($insight->getDetails());
+        self::assertNotEmpty($insight->getDetails());
+    }
+
+    public function testSkipFile(): void
+    {
+        $fileLocation = __DIR__ . '/Fixtures/ForbiddenFinalClass.php';
+
+        $collection = $this->runAnalyserOnConfig(
+            [
+                'config' => [
```

Comment:
To test it correctly without adding ForbiddenFinalClass in Architecture/Classes, I suggest to add it here locally like following: 

```suggestion
                'add' => [
                    Classes::class => [
                        ForbiddenFinalClasses::class
                    ]
                ],
                'config' => [
```